modules/shortcodes/dailymotion.php

```diff
@@ -93,40 +119,93 @@ function dailymotion_shortcode( $atts ) {
+
+	if ( isset( $atts['autoplay'] ) && '1' === $atts['autoplay'] ) {
+		$player_params['autoplay'] = '1';
+	}
+	if ( isset( $atts['endscreen-enable'] ) && '0' === $atts['endscreen-enable'] ) {
+		$player_params['endscreen-enable'] = '0';
 	}
+	if ( isset( $atts['mute'] ) && '1' === $atts['mute'] ) {
+		$player_params['mute'] = '1';
+	}
+	if ( isset( $atts['sharing-enable'] ) && '0' === $atts['sharing-enable'] ) {
+		$player_params['sharing-enable'] = '0';
+	}
+	if ( isset( $atts['start'] ) && ! empty( $atts['start'] ) ) {
+		$player_params['start'] = abs( intval( $atts['start'] ) );
+	}
+	if ( isset( $atts['subtitles-default'] ) && ! empty( $atts['subtitles-default'] ) ) {
+		$player_params['subtitles-default'] = esc_attr( $atts['subtitles-default'] );
+	}
+	if ( isset( $atts['ui-highlight'] ) && ! empty( $atts['ui-highlight'] ) ) {
+		$player_params['ui-highlight'] = esc_attr( $atts['ui-highlight'] );
+	}
+	if ( isset( $atts['ui-logo'] ) && '0' === $atts['ui-logo'] ) {
+		$player_params['ui-logo'] = '0';
+	}
+	if ( isset( $atts['ui-start-screen-info'] ) && '0' === $atts['ui-start-screen-info'] ) {
+		$player_params['ui-start-screen-info'] = '0';
+	}
+	if ( isset( $atts['ui-theme'] ) && in_array( strtolower( $atts['ui-theme'] ), array( 'dark', 'light' ) ) ) {
+		$player_params['ui-theme'] = esc_attr( $atts['ui-theme'] );
+	}
+
+	// Add those parameters to the Video URL.
+	$video_url = add_query_arg(
+		$player_params,
+		'https://www.dailymotion.com/embed/video/' . $id
+	);
 
-	$height = ( 425 == $width ) ? 334 : ( $width / 425 ) * 334;
-	$id     = urlencode( $id );
+	$output = '';
 
 	if ( preg_match( '/^[A-Za-z0-9]+$/', $id ) ) {
-		$output = '<iframe width="' . $width . '" height="' . $height . '" src="' . esc_url( '//www.dailymotion.com/embed/video/' . $id ) . '" frameborder="0"></iframe>';
-		$after  = '';
+		$output .= '<iframe width="' . esc_attr($width) . '" height="' . esc_attr($height) . '" src="' . esc_url( $video_url ) . '" style="border:0;" allowfullscreen></iframe>';
 
 		if ( array_key_exists( 'video', $atts ) && $video = preg_replace( '/[^-a-z0-9_]/i', '', $atts['video'] ) && array_key_exists( 'title', $atts ) && $title = wp_kses( $atts['title'], array() ) ) {
-			$after .= '<br /><strong><a href="' . esc_url( 'http://www.dailymotion.com/video/' . $video ) . '" target="_blank">' . esc_html( $title ) . '</a></strong>';
+			$output .= '<br /><strong><a href="' . esc_url( 'http://www.dailymotion.com/video/' . $video ) . '" target="_blank">' . esc_html( $title ) . '</a></strong>';
 		}
 
 		if ( array_key_exists( 'user', $atts ) && $user = preg_replace( '/[^-a-z0-9_]/i', '', $atts['user'] ) ) {
-			$after .= '<br /><em>Uploaded by <a href="' . esc_url( 'http://www.dailymotion.com/' . $user ) . '" target="_blank">' . esc_html( $user ) . '</a></em>';
+			$output .= '<br /><em>' . __('Uploaded by') . ' <a href="' . esc_url( 'http://www.dailymotion.com/' . $user ) . '" target="_blank">' . esc_html( $user ) . '</a></em>';
```

Comment:
You'll need to add the `jetpack` textdomain to the translation, as well as spaces, and I'd suggest switching to [`esc_html__`](https://developer.wordpress.org/reference/functions/esc_html__/):

`esc_html__( 'Uploaded by', 'jetpack' )`